daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/FrontStack.scala

```diff
@@ -76,10 +76,9 @@ final class FrontStack[+A] private (fq: FQ[A], val length: Int) {
         case FQPrepend(head, tail) =>
           if (head.length > 1) {
             Some((head.head, new FrontStack(FQPrepend(head.tail, tail), length - 1)))
-          } else if (head.length > 0) {
-            Some((head.head, new FrontStack(tail, length - 1)))
           } else {
-            throw new RuntimeException(s"FrontStack had FQPrepend with non-empty head: $head")
+            // NOTE(MH): We maintain the invariant that `head` is never empty.
```

Comment:
I'm not sure that it's helpful to know who wrote this note; if only one person understands this, it won't be merged. `git blame` will tell us if we need it.

```suggestion
            // NOTE: We maintain the invariant that `head` is never empty.
```

Reply:
Done.